### ISSUE TITLE
[FeatureRequest] Support the numpy docstring format

### ISSUE THREAD

**Issue by dhirschfeld** [pawamoy/mkdocstrings#32]
The numpy docstring format is very popular in the scientific python community.

The numpy docstring format is supported by the [`numpydoc`](https://numpydoc.readthedocs.io/en/latest/format.html) and [`napoleon`](https://www.sphinx-doc.org/en/master/usage/extensions/napoleon.html#module-sphinx.ext.napoleon) sphinx extensions.

**Comment by pawamoy** [comment]
With the numpy format, it feels like we're getting closer to rst again 😕 

But I (now) see the value of supporting multiple formats.

As said [in this comment](https://github.com/pawamoy/mkdocstrings/issues/27#issuecomment-572236277), we could implement multiple format parsers for docstrings.

I think we can also get inspiration (or re-use the code) from sphinx.ext.napoleon.

Maybe the code responsible for parsing the docstrings should also be split into another Python package. I think other projects could benefit from such a library.

Design propositions and pull requests are welcome as I don't have much time these days to tackle this 🙂

**Comment by StephenBrown2** [comment]
> With the numpy format, it feels like we're getting closer to rst again 😕

Personally I think that if one wants numpy formatting, they should use a different extension. Each one should focus on its own strengths.

**Comment by thebigmunch** [comment]
> Maybe the code responsible for parsing the docstrings should also be split into another Python package. I think other projects could benefit from such a library.

Like [this one](https://github.com/rr-/docstring_parser)? I haven't tried it out yet as I only searched for it after reading this issue but might be worth looking into.

**Comment by pawamoy** [comment]
Yes! Ideally this package would be extended to support the other requested formats as well as the one currently in use in `mkdocstrings`.

**Comment by pawamoy** [comment]
Moving this issue over pytkdocs which is now responsible for docstring parsing!
